Mathlib/Algebra/Homology/DerivedCategory/Ext/Map.lean

```diff
@@ -0,0 +1,146 @@
+/-
+Copyright (c) 2025 Nailin Guan, Jingting Wang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nailin Guan, Jingting Wang
+-/
+module
+
+public import Mathlib.Algebra.Homology.DerivedCategory.Ext.Linear
+public import Mathlib.Algebra.Homology.DerivedCategory.ExactFunctor
+
+/-!
+# Map Between Ext Induced by Exact Functor
+
+In this file, we develope the map `Ext^k (M, N) → Ext^k (F(M), F(N))`,
+where `F` is an exact functor between abelian categories.
+
+-/
+
+@[expose] public section
+
+universe w w' u u' v v'
+
+namespace CategoryTheory
+
+open Limits Abelian
+
+variable {C : Type u} [Category.{v} C] [Abelian C]
+variable {D : Type u'} [Category.{v'} D] [Abelian D]
+
+variable (F : C ⥤ D) [F.Additive] [PreservesFiniteLimits F] [PreservesFiniteColimits F]
+
+section Ext
+
+open Localization
+
+instance [h : HasExt.{w'} D] (X Y : C) : HasSmallLocalizedShiftedHom.{w'}
+    (HomologicalComplex.quasiIso D (ComplexShape.up ℤ)) ℤ
+    (((F ⋙ CochainComplex.singleFunctor D 0).obj X))
+    (((F ⋙ CochainComplex.singleFunctor D 0).obj Y)) :=
+  h (F.obj X) (F.obj Y)
+
+/-- The map between `Ext` induced by `F.mapShiftedHomAddHom`. -/
+noncomputable def Abelian.Ext.mapExactFunctor [HasExt.{w} C] [HasExt.{w'} D] {X Y : C} {n : ℕ}
+    (f : Ext.{w} X Y n) : Ext.{w'} (F.obj X) (F.obj Y) n :=
+  (F.mapHomologicalComplexUpToQuasiIsoLocalizerMorphism
+    (ComplexShape.up ℤ)).smallShiftedHomMap
+    ((F.mapCochainComplexSingleFunctor 0).app X) ((F.mapCochainComplexSingleFunctor 0).app Y) f
+
+section
+
+universe t t'
+
+variable [HasDerivedCategory.{t} C] [HasDerivedCategory.{t'} D]
```

Comment:
These variables should appear only in the statement of the lemma which really depends on them.